vignettes/baselinenowcast.Rmd

```diff
@@ -431,26 +432,72 @@ exp_obs_nowcasts <- add_obs_errors_to_nowcast(
 ```
 
 Next, we will take the list of expected observed reporting squares and
-convert this into a long tidy dataframe, indexed by reference time `t` and
-delay `d`, and plot the results
+convert this into a long tidy dataframe, indexed by reference time `t`,
+delay `d`, and draws.
 
 ```{r generate-nowcasted-draws-df}
 nowcast_draws <- nowcast_list_to_df(
   list_of_nowcasts = exp_obs_nowcasts
 )
+head(nowcast_draws)
+```
 
-summary_nowcast <- nowcast_draws |>
-  group_by(time, draw) |>
-  summarise(total_counts = sum(count))
+The package also provides a helper function, `summarise_by_reference_time()`
+which converts the draws indexed by `t` and `d` to one which contains the sum
+over all delays. We can then use this to plot the nowcasts compared to the
+data available as of the nowcast date, and the later observed data.
+```{r summarise-by-ref-time}
+summary_nowcast <- summarise_by_reference_time(nowcast_draws)
 
-ggplot(summary_nowcast) +
-  geom_line(aes(x = time, y = total_counts, group = draw),
-    alpha = 0.1, color = "gray", linewidth = 0.1
+# Join with the original data
+final_data <- data_long[location == "DE"][age_group == "00+"] |> # nolint
+  enw_filter_report_dates(latest_date = "2021-10-01") |>
+  enw_filter_reference_dates(
+    latest_date = "2021-07-01",
+    include_days = n_history_delay - 1
+  ) |>
+  group_by(reference_date) |>
```

Comment:
This info is in the preprocessed object and there is a exported function you can look at to get it.

That being said I think its fine to use dplyr here if you want to

Reply:
Ah yes, I think I should do some getting started vignette cleaning up bc I am doing a bit of repetitive data wrangling... 

My intention was to not include too much post processing in `baselinenowcast`, which is why I am just showing how you do this here vs writing an exported package function to do this (which we also could)

Or by exported function you mean from `epinowcast`?

Reply:
yes from epinowcast

Reply:
Slight preference to just use `dplyr` since it is more likely users will be familiar will easily understand what is happening. Will address general redundancy in #68